R/generate_triangles.R

```diff
@@ -25,27 +26,65 @@
 #'   byrow = TRUE
 #' )
 #'
-#' trunc_rts <- truncate_triangles(
-#'   reporting_triangle = triangle,
-#'   n = 2
-#' )
-#' retro_rts <- generate_triangles(
-#'   trunc_rep_mat_list = trunc_rts
+#' trunc_rts <- truncate_triangles(triangle, n = 2)
+#' retro_rts <- generate_triangles(trunc_rts)
+#'
+#' # With custom structure
+#' retro_rts_custom <- generate_triangles(
+#'   retro_rts,
+#'   structure = 2
 #' )
-#' print(retro_rts[[1]])
-#' print(retro_rts[[2]])
-generate_triangles <- function(trunc_rep_mat_list) {
+#' retro_rts_custom
+generate_triangles <- function(reporting_triangle_list, structure = 1) {
   # Check that input is a list of matrices
-  if (!is.matrix(trunc_rep_mat_list[[1]])) {
+  if (!is.matrix(reporting_triangle_list[[1]])) {
     cli_abort(
-      message = "The elements of `trunc_rep_mat_list`must be matrices"
+      message = "The elements of `reporting_triangle_list` must be matrices"
     )
   }
 
   results <- lapply(
-    trunc_rep_mat_list,
-    replace_lower_right_with_NA
+    reporting_triangle_list,
+    replace_lower_right_with_NA,
+    structure = structure
   )
 
   return(results)
 }
+
+#' Generate a single retrospective reporting triangle
+#'
+#' This function generates a single reporting triangle by removing the bottom
+#'   right observations from a truncated reporting triangle matrix. It is the
+#'   singular version of `generate_triangles()`.
+#'
+#' @param reporting_triangle A single reporting triangle matrix.
+#' @inheritParams replace_lower_right_with_NA
+#' @returns A single retrospective reporting triangle matrix with NAs in the
+#'   appropriate positions.
+#' @export
+#' @examples
+#' triangle <- matrix(
+#'   c(
+#'     65, 46, 21, 7,
+#'     70, 40, 20, 5,
+#'     80, 50, 10, 10,
+#'     100, 40, 31, 20,
+#'     95, 45, 21, NA,
+#'     82, 42, NA, NA,
+#'     70, NA, NA, NA
+#'   ),
+#'   nrow = 7,
+#'   byrow = TRUE
+#' )
+#'
+#' retro_rt <- generate_triangle(triangle)
+#' retro_rt
+generate_triangle <- function(reporting_triangle, structure = 1) {
```

Comment:
I suggest we make .replace.... internal and suggest this for users

Reply:
I am suggesting we remove .replace entirely and use this interface but this was a significant code change we didn't explicitly need here. I think it should be addressed in a new issue